packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.test.tsx

```diff
@@ -81,4 +82,67 @@ describe('useFluentProvider_unstable', () => {
       }
     `);
   });
+
+  it('should merge customStyles', () => {
+    const customStylesA: FluentProviderCustomStyleHooks = {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      useButtonStyles_unstable: (state: unknown) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (state as any).testResult = 'useButtonStyles_unstable_A';
+      },
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      useCompoundButtonStyles_unstable: (state: unknown) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (state as any).testResult = 'useCompoundButtonStyles_unstable_A';
+      },
+    };
+
+    const customStylesB: FluentProviderCustomStyleHooks = {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      useCompoundButtonStyles_unstable: (state: unknown) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (state as any).testResult = 'useCompoundButtonStyles_unstable_B';
+      },
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      useSplitButtonStyles_unstable: (state: unknown) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (state as any).testResult = 'useSplitButtonStyles_unstable_B';
+      },
+    };
```

Comment:
nit: You don't need `state` to be used there actually, you can just return a string:

```suggestion
    const customStylesA: FluentProviderCustomStyleHooks = {
      // eslint-disable-next-line @typescript-eslint/naming-convention
      useButtonStyles_unstable: () => {
        return 'useButtonStyles_unstable_A';
      },
      // eslint-disable-next-line @typescript-eslint/naming-convention
      useCompoundButtonStyles_unstable: () => {
        return 'useCompoundButtonStyles_unstable_A';
      },
    };

    const customStylesB: FluentProviderCustomStyleHooks = {
      // eslint-disable-next-line @typescript-eslint/naming-convention
      useCompoundButtonStyles_unstable: () => {
        return 'useCompoundButtonStyles_unstable_B';
      },
      // eslint-disable-next-line @typescript-eslint/naming-convention
      useSplitButtonStyles_unstable: () => {
        return 'useSplitButtonStyles_unstable_B';
      },
    };
```